### ISSUE TITLE
NaN return 

### ISSUE THREAD

**Issue by chase26091** [libgdx/gdx-ai#45]
```
if (steeringBehavior != null) {
        steeringBehavior.calculateSteering(steeringOutput);
        if (this.body != null && !steeringOutput.linear.isZero()) {
            System.out.println(steeringOutput.linear);
            this.body.applyForceToCenter(steeringOutput.linear, true);
        }
        // applySteering(steeringOutput, deltaTime);
    }
```

I'm using gdx-ai 1.5.0

calculateSteering is returning a NaN,NaN vector. 
I am using the example Box2dRadiusProximity with the CollisionAvoidance steering behavior. No errors are thrown, my box2d body just disappears once i approach the radius distance, very weird behavior. I don't understand where the issue is spawning from

**Comment by davebaol** [comment]
Please provide a [SSCE](http://sscce.org/). It's your best bet to get it fixed.
Also, you can try gdxai-1.5.1-SNAPSHOT

**Comment by davebaol** [comment]
Please try gdxai-1.6.0

**Comment by davebaol** [comment]
Is this still an actual issue?
I'll close it if I do not get a feedback within a few days.

**Comment by davebaol** [comment]
Ok, closing this since without SCCE there's no way I can reproduce your issue.
Please reopen if you can supply the SSCE.

**Comment by lukz** [comment]
I've got same issue. It's happening in Separation behavior.

`	@Override
	public boolean reportNeighbor (Steerable<T> neighbor) {

		toAgent.set(owner.getPosition()).sub(neighbor.getPosition());
		float distanceSqr = toAgent.len2();
		float maxAcceleration = getActualLimiter().getMaxLinearAcceleration();

		// Calculate the strength of repulsion through inverse square law decay
		float strength = getDecayCoefficient() / distanceSqr;
		if (strength > maxAcceleration) strength = maxAcceleration;

		// Add the acceleration
		// Optimized code for linear.mulAdd(toAgent.nor(), strength);
		linear.mulAdd(toAgent, strength / (float)Math.sqrt(distanceSqr));

		return true;
	}`

Basically distanceSqr can be 0 and in this case we've division by zero.

**Comment by davebaol** [comment]
@lukz 
Correct. Reopening this.
Any chance of a PR?

**Comment by lukz** [comment]
@davebaol I'm not sure what's the correct way to handle that. In my case I'm just checking steeringOutput after calculate steering if it's NaN and setting it to zero. But it's just temporary hack.

Maybe if distanceSqr == 0 than setting it to Float.MIN_VALUE is the way to go?

**Comment by davebaol** [comment]
@lukz

> Maybe if distanceSqr == 0 than setting it to Float.MIN_VALUE is the way to go?

Even if you do this the produced linear acceleration would not change because you mulAdd toAgent, which is a zero vector. I'd say that we could just return true in this case.
